sycl/include/sycl/ext/intel/esimd/memory.hpp

```diff
@@ -266,127 +223,21 @@ scatter(Tx *p, Toffset offset, simd<Tx, N> vals, simd_mask<N> mask = 1) {
-#else
-                                   uint32_t offset,
-#endif
-                                   Flags flags = {}) {
+namespace detail {
+// Accessors may get either 32-bit offset or 64-bit depending on
+// the -fsycl-esimd-force-stateles-mem mode settigs.
 #ifdef __ESIMD_FORCE_STATELESS_MEM
-  return block_load<Tx, N>(__ESIMD_DNS::accessorToPointer<Tx>(acc, offset),
-                           flags);
+using DeviceAccessorOffsetT = uint64_t;
 #else
-  std::ignore = flags;
-  constexpr unsigned Sz = sizeof(T) * N;
-  static_assert(Sz >= detail::OperandSize::OWORD,
-                "block size must be at least 1 oword");
-  static_assert(Sz % detail::OperandSize::OWORD == 0,
-                "block size must be whole number of owords");
-  static_assert(detail::isPowerOf2(Sz / detail::OperandSize::OWORD),
-                "block must be 1, 2, 4 or 8 owords long");
-  static_assert(Sz <= 8 * detail::OperandSize::OWORD,
-                "block size must be at most 8 owords");
-
-  auto surf_ind = __esimd_get_surface_index(
-      detail::AccessorPrivateProxy::getQualifiedPtrOrImageObj(acc));
-
-  if constexpr (Flags::template alignment<simd<T, N>> >=
-                detail::OperandSize::OWORD) {
-    return __esimd_oword_ld<T, N>(surf_ind, offset >> 4);
-  } else {
-    return __esimd_oword_ld_unaligned<T, N>(surf_ind, offset);
-  }
+using DeviceAccessorOffsetT = uint32_t;
 #endif
-}
 
-/// Stores elements of the vector \p vals to a contiguous block of memory
-/// at the given address \p addr.
-/// The generated code depends on the combination {T, N, Flags}.
-/// Providing flags specifying the alignment of 16-bytes or more produces more
-/// efficient code. If the alignment is smaller than 16-bytes, then less
-/// efficient scatter is generated. If the stored vector is too long
-/// for 1 flat-store GPU instruction, then a series of flat-store and/or
-/// scatters may be generated.
-/// @tparam Tx Element type.
-/// @tparam N Number of elements to store.
-/// @tparam Flags The alignment specifier type tag.
-/// @param addr The memory address to store at.
-/// @param vals The vector to store.
-/// @param Flags Specifies the alignment.
-template <typename Tx, int N,
-          typename Flags = overaligned_tag<detail::OperandSize::OWORD>>
-__ESIMD_API std::enable_if_t<is_simd_flag_type_v<Flags>>
-block_store(Tx *addr, simd<Tx, N> vals, Flags = {}) {
-  using T = typename detail::__raw_t<Tx>;
-  using VecT = typename simd<T, N>::raw_vector_type;
-  constexpr size_t Align = Flags::template alignment<simd<T, N>>;
-  __esimd_svm_block_st<T, N, Align>(reinterpret_cast<VecT *>(addr),
-                                    vals.data());
-}
-
-namespace detail {
-template <
-    typename T, int NElts,
-    __ESIMD_DNS::lsc_data_size DS = __ESIMD_DNS::lsc_data_size::default_size,
-    cache_hint L1H = cache_hint::none, cache_hint L2H = cache_hint::none,
-    typename FlagsT>
+template <typename T, int NElts, cache_hint L1H = cache_hint::none,
```

Comment:
this block_load_impl() function is actually the copy of the old experimental::lsc_block_load().
Please pay attention that this function here doesn't have the 'lsc_data_size' template parameter that was used in the old experimental::lsc_block_load().
The reason for that is that the data-size is always deduced from the loaded element type.